### PR TITLE
Timepicker single & validation

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

- rename time-range-input -> timepicker
- Add support for single time input
- Add validation for time picker
- Add base-class for date- and time-picker to reduce code duplications

## References

- #419 #513

### CODE REVIEW

**Review by ManBearTM — DISMISSED** [review]
changes
All that with -10 lines of code 👏

**Review by Renetnielsen — APPROVED** [review]

File: libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+    this.stateChanges.complete();
+  }
+
+  protected abstract initRangeInput(): void;
+  protected abstract initSingleInput(): void;
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts:263]: suggestion: add ignore tags

File: libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+  };
+
+  /**
+   * @ignore
+   */
+  protected markParentControlAsTouched = (): void => {
+    // Intentionally left empty
+  };
+
+  private setRangeValue(
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts:348]: suggestion: add ignore tag

File: libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    if (start) {
+      startInput.value = start;
+    }
+
+    if (end) {
+      endInput.value = end;
+    }
+  }
+
+  private setSingleValue(value: string, input: HTMLInputElement) {
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts:364]: suggestion: add ignore tag

File: libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+  /**
+   * @ignore
+   */
+  set placeholder(value: string) {
+    this._placeholder = value;
+    this.stateChanges.next();
+  }
+
+  protected abstract _placeholder: string;
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/input/shared/watt-picker-base.ts:111]: suggestion: add ignore tag

**Review by dzhavat — CHANGES_REQUESTED** [review]

**Review by dzhavat — DISMISSED** [review]

**Review by dzhavat — APPROVED** [review]

**Pull request merged**